pandas/tests/frame/methods/test_set_axis.py

```diff
@@ -27,18 +27,21 @@ def test_set_axis(self, obj):
     @pytest.mark.parametrize("axis", [0, "index", 1, "columns"])
     def test_set_axis_inplace_axis(self, axis, obj):
         # GH#14636
-        if obj.ndim == 1 and axis in [1, "columns"]:
+        axis = obj._get_axis_number(axis)
+        if obj.ndim == 1 and (axis == "columns" or axis == 1):
```

Comment:
on line 30 `axis` is set to an `int` and therefore on line 31 `axis == "columns"` is impossible.